third_party/terraform/tests/resource_storage_bucket_object_test.go

```diff
@@ -390,6 +422,24 @@ resource "google_storage_bucket_object" "object" {
 `, bucketName, objectName, content, storageClass)
 }
 
+func testGoogleStorageBucketsObject_metadata(bucketName string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+  name = "%s"
+}
+
+resource "google_storage_bucket_object" "object" {
+  name          = "%s"
+  bucket        = google_storage_bucket.bucket.name
+  content       = "%s"
+
+  metadata = {
+    "customKey" = "custom_value"
```

Comment:
is the quoted key valid HCL?

Reply:
Yup!  It's valid.  I just ran `0.12upgrade` on it too and it didn't change it, so I think it's fine?  Either way works, though, so if you'd prefer I change it, I can!